server/src/controllers/auth.controller.ts

```diff
@@ -74,4 +76,24 @@ export class AuthController {
       ImmichCookie.IS_AUTHENTICATED,
     ]);
   }
+
+  @Post('create-pincode')
+  @HttpCode(HttpStatus.CREATED)
+  @Authenticated()
+  createPincode(@Auth() auth: AuthDto, @Body() dto: createPincodeDto): Promise<UserAdminResponseDto> {
+    return this.service.createPincode(auth, dto);
+  }
+
+  @Post('change-pincode')
+  @HttpCode(HttpStatus.OK)
+  @Authenticated()
+  changePincode(@Auth() auth: AuthDto, @Body() dto: ChangePincodeDto): Promise<UserAdminResponseDto> {
+    return this.service.changePincode(auth, dto);
+  }
+
+  @Get('has-pincode')
+  @Authenticated()
+  hasPincode(@Auth() auth: AuthDto): Promise<boolean> {
+    return this.service.hasPincode(auth);
+  }
```

Comment:
I am still not 100% convinced that those should be here and not in the user controller tbh. In my mind it's very much user-oriented/scoped.